[1.4] Fully unload Jofairden armor assets

ExterminatorX99 opened the pull request:
### What is the bug?
#2902 

### How did you fix the bug?
Fully unload Jofairden armor assets

### Are there alternatives to your fix?
Change `??=` to `if` and add a disposed check

ExterminatorX99 committed:
Fully unload Jofairden armor assets

Fixes #2902

Chicken-Bones commented:
This fix is correct, but, in my opinion, a better fix would be to remove the `static` modifier from these fields. The layers are re-created each time the mod is reloaded, so new assets will be procured correctly

ExterminatorX99 committed:
Remove the `static` modifier from the fields

The layers are re-created each reload, so new assets will be procured correctly
